### PR TITLE
chore(api-client): remove axios again

### PR DESCRIPTION
**Pull request by hanspagel**
WIP

**Tasks**

- [x] Make the tests run
- [x] Make the UI work
- [x] Fix related TS errors
- [x] allow non-proxy calls
- [x] Check whether we can abort a request
- [x] Setting server path works ex `https://galaxy.scalar.com/api`
- [x] fix variable replacement
- [x] add :colon variables
- [x] add error handling
- [x] make cookies work - not quite sure how to test cookies, but made it stop erroring when URL creation is failing

Then review, approve and merge :)

### CODE REVIEW

File: packages/api-client/src/libs/send-request.ts [hunk 1]
```diff
@@ -278,36 +290,25 @@ export function createRequestOperation({
...
-
-      headersToRemove
-        .map((header) => header.toLowerCase())
-        .forEach((header) => response.headers.delete(header))
-    }
-
+    const responseHeaders = normalizeHeaders(
+      response.headers,
+      shouldUseProxy(proxy, url.origin),
+    )
```
Comment (hanspagel) [inline packages/api-client/src/libs/send-request.ts:296]: note to myself: make sure this is still in the code

File: packages/api-client/src/views/Request/Request.vue [hunk 2]
```diff
@@ -66,39 +67,24 @@ const executeRequest = async () => {
...
+
+  // Send error toast
+  if (sendRequestError) toast(sendRequestError.message, 'error')
+  else requestHistory.push(result)
 }
 
 /** Cancel a live request */
-const cancelRequest = async () => requestAbortController.value?.abort()
+const cancelRequest = async () =>
+  requestAbortController.value?.abort('User has cancelled the request')
```
Comment (antlio) [inline packages/api-client/src/views/Request/Request.vue:86]: what do you think about that as the action is triggered by the user?

```suggestion
  requestAbortController.value?.abort('The request has been cancelled')
```

Comment (amritk) [reply packages/api-client/src/views/Request/Request.vue:86]: Sounds good to me!

**Pull request merged**